src/dune_rules/dune_package.ml

```diff
@@ -458,12 +458,17 @@ module Or_meta = struct
     let dir = Path.parent_exn file in
     Path.as_outside_build_dir_exn file
     |> Fs_memo.with_lexbuf_from_file ~f:(fun lexbuf ->
-           (* XXX stop catching code errors, invalid args, etc. *)
-           Result.try_with (fun () ->
-               Vfile.parse_contents lexbuf ~f:(fun lang ->
-                   String_with_vars.set_decoding_env
-                     (Pform.Env.initial lang.version)
-                     (decode ~lang ~dir))))
+           match
+             Vfile.parse_contents lexbuf ~f:(fun lang ->
+                 String_with_vars.set_decoding_env
+                   (Pform.Env.initial lang.version)
+                   (decode ~lang ~dir))
+           with
+           | x -> Ok x
+           | exception User_error.E message -> Error message
+           | exception exn ->
+             (* XXX stop catching code errors, invalid args, etc. *)
```

Comment:
You could take the opportunity to stop catching this stuff error. Off the top of my head, you only need to catch Sys_error, and End_of_file errors here.

Reply:
Will do a more thorough analysis and fix in a follow-up PR.